src/aurora/gff3writer.cpp

```diff
@@ -0,0 +1,610 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  Writer for writing version 3.x of biowares gff (general file format).
+ */
+
+#include <algorithm>
+
+#include "src/common/writestream.h"
+#include "src/common/memreadstream.h"
+
+#include "src/aurora/gff3writer.h"
+
+namespace Aurora {
+
+GFF3Writer::GFF3Writer(uint32 id, uint32 version) : _id(id), _version(version) {
+	_structs.push_back(GFF3WriterStructPtr(new GFF3WriterStruct(this)));
```

Comment:
Can you use boost::make_shared instead of `new`'ing the struct into the shared_ptr (here and everywhere else as well)?

I.e. `boost::make_shared<GFF3WriterStruct>(this)` should be equivalent to `boost::shared_ptr<GFF3WriterStruct>(new GFF3WriterStruct(this))`, and gets rid of the explicit `new`.